packages/server/src/unstable-core-do-not-import/procedureBuilder.ts

```diff
@@ -93,10 +93,10 @@ export type AnyProcedureBuilder = ProcedureBuilder<
   any,
   any,
   any,
-  any,
-  any,
-  any,
-  any
+  UnsetMarker,
+  UnsetMarker,
+  UnsetMarker,
+  UnsetMarker
```

Comment:
So I'm basically 100% sure this is going to cause regressions which out test suite may not be capturing yet (because we never broke it before)

Here's the reason: https://trpc.io/docs/server/validators#input-merging

and here's the scenario:

<img width="811" alt="image" src="https://github.com/trpc/trpc/assets/8896153/2dc207b8-7317-44bd-a446-d7d5d3ddaf71">

It's totally valid to set up an input or output and then create a router with that procedure, so UnsetMarker wouldn't be the root type in that situation

Reply:
Okay so as per https://github.com/trpc/trpc/pull/5497/files#r1494523733 I think what we could do is leave _this_ AnyProcedureBuilder be

If you copy it over to packages/react-query/src/shared/polymorphism and export it from there though, I think we can live with its shortcomings as a limitation and open it up later. But we shouldn't make the rest of the codebase live with UnsetMarker hard-coded